addOns/openapi/src/main/java/org/zaproxy/zap/extension/openapi/OpenApiAPI.java

```diff
@@ -129,4 +138,24 @@ public ApiResponse handleApiAction(String name, JSONObject params) throws ApiExc
             throw new ApiException(ApiException.Type.BAD_ACTION);
         }
     }
+
+    private Integer getContextId(JSONObject params) throws ApiException {
+        int ctxId;
+        try {
+            if (params.containsKey(PARAM_CONTEXT_ID)
+                    && !params.getString(PARAM_CONTEXT_ID).isEmpty()) {
+                ctxId = ApiUtils.getContextByParamId(params, PARAM_CONTEXT_ID).getId();
+            } else {
+                List<Context> contexts = Model.getSingleton().getSession().getContexts();
+                if (contexts.size() > 0) {
+                    ctxId = contexts.get(0).getId();
+                } else {
+                    ctxId = -1;
+                }
+            }
+        } catch (ApiException e) {
+            throw new ApiException(ApiException.Type.ILLEGAL_PARAMETER, PARAM_TARGET);
```

Comment:
The catch should be removed, the above method already throws the expected exception.